### PR TITLE
Fuzz coverage recording

### PR DESCRIPTION
**Pull request by tevoinea**
## Summary of the Pull Request

_What is this about?_

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

File: src/agent/coverage/fuzz/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# Prevent this from interfering with workspaces
+[workspace]
+members = ["."]
+
+[profile.release]
+debug = 1
+
+[[bin]]
+name = "fuzz_target_1"
```
Comment (mgreisen) [inline src/agent/coverage/fuzz/Cargo.toml:27]: I think you have to change the name and target values to match your actual filename.

Comment (tevoinea) [reply src/agent/coverage/fuzz/Cargo.toml:27]: Good catch!

File: src/agent/coverage/fuzz/fuzz_targets/fuzz_target_record_coverage.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        return;
+    }
+
+    // Write mutated bytes to a file
+    let mut file = NamedTempFile::new_in(env::current_dir().unwrap()).unwrap();
+    file.write_all(data);
+    let path = String::from(file.path().to_str().unwrap());
+
+    // Make sure the file is executable
+    Command::new("chmod").args(["+wrx", &path]).spawn().unwrap();
```
Comment (Porges) [inline src/agent/coverage/fuzz/fuzz_targets/fuzz_target_record_coverage.rs:32]: This should be waited upon to finish:
```suggestion
    Command::new("chmod").args(["+wrx", &path]).spawn().unwrap().wait();
```

I think also that once this is done you don't need the `file.keep` or `fs::remove_file` calls; the file should be kept until the end of the function.

Comment (tevoinea) [reply src/agent/coverage/fuzz/fuzz_targets/fuzz_target_record_coverage.rs:32]: The reason I had `file.keep`/`fs::remove_file` was that tempfile kept a handle open on the file (and deleting the file on `drop`) which was causing issues with our binary inspection.

**Review by mgreisen — APPROVED** [review]

**Pull request merged**